plugins/allocators/bitmap/bitmap_ipv4.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2018-present the CoreDHCP Authors. All rights reserved
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package bitmap
+
+// This allocator handles IPv4 assignments with a similar logic to the base bitmap, but a simpler
+// implementation due to the ability to just use uint32 for IPv4 addresses
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"net"
+	"sync"
+
+	"github.com/coredhcp/coredhcp/plugins/allocators"
+	"github.com/willf/bitset"
+)
+
+var (
+	// ErrNoAddrAvail is returned when we can't allocate an IP
+	// because there's no unallocated space left
+	ErrNoAddrAvail = errors.New("No address available to allocate")
+
+	errNotInRange = errors.New("IP outside of allowed range")
+
+	errInvalidIP = errors.New("Invalid IP passed as input")
+)
+
+// IPv4Allocator allocates IPv4 addresses, tracking utilization with a bitmap
+type IPv4Allocator struct {
+	start uint32
+	end   uint32
+
+	bitmap *bitset.BitSet
+	l      sync.Mutex
```

Comment:
I'll move this to use a concurrent bitset at some point in the future, https://github.com/tmthrgd/go-bitset seemed nice for example, so that we can remove the mutex here (and in the v6 allocator as well)